lib/rules/no-sync.js

```diff
@@ -19,22 +19,79 @@ module.exports = {
             recommended: false
         },
 
-        schema: []
+        schema: [
+            {
+                type: "object",
+                properties: {
+                    allowAtRootLevel: {
+                        type: "boolean"
+                    }
+                },
+                additionalProperties: false
+            }
+        ]
     },
 
     create(context) {
+        const allowAtRootLevel = context.options[0] ? context.options[0].allowAtRootLevel : false;
+        const stackOfSyncExpressions = [];
+        let setOfSyncExpressions = new Set();
 
-        return {
+        /**
+         * push Sync Expression sets and track Sync Expressions in a new set
+         *
+         * @returns {void}
+         */
+        function trackSyncExpression() {
+            if (allowAtRootLevel) {
+                stackOfSyncExpressions.push(setOfSyncExpressions);
+                setOfSyncExpressions = new Set();
+            }
+        }
 
-            "MemberExpression[property.name=/.*Sync$/]"(node) {
-                context.report({
-                    node,
-                    message: "Unexpected sync method: '{{propertyName}}'.",
-                    data: {
-                        propertyName: node.property.name
-                    }
-                });
+        /**
+         * Report if the Sync Expression is not in the root level
+         *
+         * @param {ASTNode} node the FunctionDeclaration/Expression node
+         * @returns {void}
+         */
+        function reportIfSyncInFunction(node) {
+            if (allowAtRootLevel) {
+                if (setOfSyncExpressions.size > 0) {
+                    setOfSyncExpressions.forEach(syncExpression => {
+                        context.report({
+                            node,
+                            message: "Unexpected sync method: '{{propertyName}}'.",
+                            data: {
+                                propertyName: syncExpression.property.name
+                            }
+                        });
+                    });
+
+                }
+                setOfSyncExpressions.clear();
+                setOfSyncExpressions = stackOfSyncExpressions.pop();
             }
+        }
+
+        return {
+            "MemberExpression[property.name=/.*Sync$/]"(node) {
```

Comment:
that's what I wanted to do, but didn't know how to. Will know better next time!
@not-an-aardvark I fixed with your changes